contracts/vaults/Deposits.sol

```diff
@@ -27,7 +27,10 @@ contract Deposits is Base {
         if (shares == 0) revert AmountToDepositIsZero();
         // Users can only deposit for themselves if withdraw request exists
         if (withdrawEnableStartTime[receiver] != 0 && receiver != caller) {
-            revert CannotDepositToAnotherUserWithWithdrawRequest();
+            HATVaultsRegistry.GeneralParameters memory generalParameters = registry.getGeneralParameters();
```

Comment:
so:

a) seems to me this check is not needed, as `super._deposit(.. receiver..)` will call `_beforeTokenTransfer` and do the check

b) are you sure users can deposit for themselves if a withdraw request exists? Seems to me it would fail on the new _beforeTokenTransfer (I could not find a test, but it is hard to find tests :)

REcommendation :) remove the lines here in deposit and handle all the logic in _beforeTokentransfer

Reply:
The way I implemented the _beforeTokentransfer is that it is ignored for both mint and burn operations so it does not apply here. You think I should change that?

Reply:
ok I see. It seems cleaner to do it in one place.